### ISSUE TITLE
Exception in finalizer

### ISSUE THREAD

**Issue by JoeSSS** [mat813/rb-kqueue#17]
After #15 we started to get: 
```
Exception in finalizer #<Proc:0x0000000108289a20 ...ruby-3.1.0/gems/rb-kqueue-0.2.7/lib/rb-kqueue/watcher/file.rb:45 (lambda)>
...rb-kqueue-0.2.7/lib/rb-kqueue/watcher/file.rb:45:in `block in finalizer': wrong number of arguments (given 1, expected 0) (ArgumentError)
```

We run kqueue like this: 

```
@thread = Thread.new do
  Thread.current.abort_on_exception = true
  queue = KQueue::Queue.new
  queue.watch_file(logfile, :extend) do
      operation_on(file.read)
   end
   queue.run
end
```

when we kill the thread `Thread.kill(@thread)` the exception is raised and since we use `abort_on_exception = true`, this results in crashes in unrelated part of code of other gems.

Most likely we do something weird here, would be happy to hear any suggestions.

**Comment by JoeSSS** [comment]
It was reproducible on ruby 2.7.1 too, so I believe it is not related to Ruby 3

**Comment by JoeSSS** [comment]
@marvinthepa can you maybe take a look, since you definitely understand the finalizer better than I do :)

**Comment by dsh2dsh** [comment]
I've just got the same issue during upgrade from ruby 3.0 to 3.1

**Comment by dsh2dsh** [comment]
If it helps, I found it in ri ObjectSpace.define_finalizer

> If aProc is a lambda or method, make sure it can be called with a single argument.

**Comment by marvinthepa** [comment]
> @marvinthepa can you maybe take a look, since you definitely understand the finalizer better than I do :)

Sorry, I was indisposed. Also sorry for this mistake in the finalizer.

#18 looks good to me, although I also would make the argument optional (as it says "make sure it _can_ be called with a single argument").

**Comment by dsh2dsh** [comment]
@marvinthepa  thanks, I've made it optional.

**Comment by gregschmit** [comment]
Merged and released as `0.2.8`.
